src/platform/plugins/shared/discover/public/application/main/discover_main_route.test.tsx

```diff
@@ -77,120 +145,150 @@ describe('DiscoverMainRoute', () => {
 
-    await waitFor(() => {
-      component.update();
-      expect(findTestSubject(component, 'kbnNoDataPage').length).toBe(1);
-    });
+    await waitForLoad();
+
+    expect(screen.getByTestId('kbnNoDataPage')).toBeVisible();
   });
 
   test('renders no data view when hasESData=true & hasUserDataView=false', async () => {
-    const component = mountComponent(true, false);
-
-    await waitFor(() => {
-      component.update();
-      expect(findTestSubject(component, 'noDataViewsPrompt').length).toBe(1);
-    });
-  });
+    setupComponent({ hasESData: true, hasUserDataView: false });
 
-  // skipped because this is the case that never ever should happen, it happened once and was fixed in
-  // https://github.com/elastic/kibana/pull/137824
-  test.skip('renders no data page when hasESData=false & hasUserDataView=true', async () => {
-    const component = mountComponent(false, true);
+    await waitForLoad();
 
-    await waitFor(() => {
-      component.update();
-      expect(findTestSubject(component, 'kbnNoDataPage').length).toBe(1);
-    });
+    expect(screen.getByTestId('noDataViewsPrompt')).toBeVisible();
   });
 
   test('renders LoadingIndicator while customizations are loading', async () => {
     let resolveService = (_: DiscoverCustomizationService) => {};
     mockCustomizationService = new Promise((resolve) => {
       resolveService = resolve;
     });
-    const component = mountComponent(true, true);
-    await waitFor(() => {
-      component.update();
-      expect(component.find(DiscoverMainApp).exists()).toBe(false);
-    });
+    setupComponent({ hasESData: true, hasUserDataView: true });
+
+    expect(screen.getByLabelText('Loading')).toBeInTheDocument();
+
     resolveService(createCustomizationService());
-    await waitFor(() => {
-      component.setProps({}).update();
-      expect(component.find(DiscoverMainApp).exists()).toBe(true);
-    });
+    await waitForLoad();
+
+    expect(screen.queryByLabelText('Loading')).not.toBeInTheDocument();
   });
 
   test('renders LoadingIndicator while root profile is loading', async () => {
     mockRootProfileState = { rootProfileLoading: true };
-    const component = mountComponent(true, true);
-    await waitFor(() => {
-      component.update();
-      expect(component.find(DiscoverMainApp).exists()).toBe(false);
-    });
-    mockRootProfileState = defaultRootProfileState;
-    await waitFor(() => {
-      component.setProps({}).update();
-      expect(component.find(DiscoverMainApp).exists()).toBe(true);
+
+    setupComponent({ hasESData: true, hasUserDataView: true });
+
+    expect(screen.getByLabelText('Loading')).toBeInTheDocument();
+  });
+
+  describe.each([
+    { hasChanged: false, id: undefined },
+    { hasChanged: true, id: undefined },
+    { hasChanged: false, id: '1234' },
+  ])('when there are no unsaved changes', ({ hasChanged, id }) => {
+    it('should call the default action', () => {
```

Comment:
Can we differentiate the descriptions so it's clear what each is doing when running the tests?